<launch>

    <node pkg="prediction_kf" type="prediction_kf_node" name="prediction_kf_node" output="screen">
        <param name="visual_topic" type="string" value="/pnp_twist"/>
        <param name="real_visual_topic" type="string" value="/detected_armor"/>
        <param name="publisher_topic" type="string" value="/prediction_kf/predict"/>
        <param name="debug_topic" type="string" value="/prediction_kf/preprocessed"/>
        <param name="R_pos" type="double" value="8.0"/>
        <param name="Q_pos" type="double" value="1.0"/>
        <param name="Q_vel" type="double" value="128.0"/>
        <param name="P_matrix_weight" type="double" value="65.0"/>
        <param name="chi_square_threshold" type="double" value="1000.0"/>
        <param name="outlier_l2_norm_ratio" type="double" value="1.5"/>
        <param name="repropagate_time" type="int" value="0"/>
    </node>

    <node pkg="rotation_ekf" type="rotation_ekf_node" name="rotation_ekf_node" output="screen">
        <param name="predict_result" type="string" value="/prediction_kf/predict"/>
        <param name="preprocessed_result" type="string" value="/prediction_kf/preprocessed"/>
        <param name="publisher_topic" type="string" value="/rotation_ekf/filtered"/>
        <param name="predict_topic" type="string" value="/rotation_ekf/predict"/>
        <param name="preprocess_topic" type="string" value="/rotation_ekf/preprocess"/>
        <param name="imu_topic" type="string" value="/can_receive_node/imu_16470"/>
    </node>

    <node pkg="rosbag" type="play" name="static_target_moving_chassis" output="screen"
          args="-s 0 -u 25 /home/ros/Downloads/2018-07-11-22-28-44.bag -q"/>
    <!--args="-s 0 -u 10 /home/ros/Downloads/2018-07-10-14-25-42.bag -q"/>-->
    <!--args="-s 0 /home/ros/Downloads/2018-06-21-12-11-35.bag -q"/>-->
</launch>
